src/py42/services/cases.py

```diff
@@ -0,0 +1,196 @@
+            page_number (int, optional): Page number of the results. Defaults to 1.
+            page_size (int, optional): Number of results to return per page. Defaults to 500.
+            sort_direction (str, optional): The direction on which to sort the response,
+                based on the corresponding sort key. `asc` or `desc`. Defaults to `asc`.
+            sort_key (str, optional): Values on which the response will be sorted. Defaults to "number".
+                Available options are name, number, createdAt, updatedAt, status, assigneeUsername, subjectUsername.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response` objects
+            that each contain a page of cases.
+        """
+        return get_all_pages(
+            self.get_page,
+            u"cases",
+            name=name,
+            status=status,
+            created_at=created_at,
+            updated_at=updated_at,
+            assignee=assignee,
+            subject=subject,
+            page_number=page_number,
+            page_size=page_size,
+            sort_direction=sort_direction,
+            sort_key=sort_key,
+            **kwargs
+        )
+
+    def get_case(self, case_number):
+        """Retrieve case details by case number.
+
+        Args:
+            case_number (int): Case number of the case.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        return self._connection.get("{}/{}".format(self._uri_prefix, case_number))
+
+    def export_summary(self, case_number):
+        """Download case summary in a PDF file.
+
+        Args:
+            case_number (int): Case number of the case.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        uri_prefix = u"{}/{}/{}".format(self._uri_prefix, case_number, u"export")
+        return self._connection.get(uri_prefix)
+
+    def update(
+        self,
+        case_number,
+        name=None,
+        subject=None,
+        assignee=None,
+        description=None,
+        findings=None,
+    ):
+        """Update case details for the given case number.
+
+        Args:
+            case_number (int): Case number of the case.
+            name (str, optional): Name of the case. Defaults to empty string.
+            subject (str, optional): A subject of the case. Defaults to empty string.
+            assignee (str, optional): User UID of the assignee. Defaults to empty string.
+            description (str, optional): Description of the case. Defaults to empty string.
+            findings (str, optional): Notes on the case. Defaults to empty string.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+
+        data = {
+            u"name": name,
+            u"assignee": assignee,
+            u"description": description,
+            u"findings": findings,
+            u"subject": subject,
+        }
+        return self._connection.put(
```

Comment:
talked offline - handle when trying to update a closed case via something like 

```
      except Py42BadRequestError as err:
            if u"NO_EDITS_ONCE_CLOSED" in err.response.text:
                raise Exception("No edits once closed")
            raise
```

Except using a custom exception

Reply:
Added.